### PR TITLE
Import and use algorithms like mantid simpleapi

### PR DESCRIPTION
**Pull request by rboston628**
## Description of work

This updates the `__init__.py` in the `algorithm` module so that algorithms are automatically put into the mantid simpleapi.

This means all algorithms can now be called as functions.

This also means MantidSnapper can easily import all of the algorithms, so that it is no longer needed to import an algorithm before using it with MantidSnapper.  This cleans up a lot of imports in files.

## Explanation of work

Created the `__init__.py` file to pull in all modules, initialize them, then put them into the simpleapi.

They can then be imported as follows:
``` python
import snapred.backend.recipe.algorithm
from mantid.simpleapi import WhateverAlgorithm
```

### Dev testing

Run the following (can run as command line script)
``` python
import pytest

with pytest.raises(ImportError):
    from mantid.simpleapi import Utensils

with pytest.raises(NameError):
    Utensils()

import snapred.backend.recipe.algorithm
from mantid.simpleapi import Utensils
Utensils()

```

### CIS testing

``` python
# replace with your test script below
```

## Link to EWM item

[EWM#<ticket_number>](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=<ticket_number>)

### CODE REVIEW

File: tests/unit/backend/recipe/algorithm/test_CalibrationMetricExtractionAlgorithm.py
Comment (rboston628) [inline tests/unit/backend/recipe/algorithm/test_CalibrationMetricExtractionAlgorithm.py:1]: This had a patch that created an error.  But also the patch was not necessary nor used.

File: src/snapred/backend/recipe/algorithm/__init__.py
Comment (rboston628) [inline src/snapred/backend/recipe/algorithm/__init__.py:1]: This is the main edit.

File: src/snapred/backend/recipe/algorithm/CalculateDiffCalTable.py
Comment (rboston628) [inline src/snapred/backend/recipe/algorithm/CalculateDiffCalTable.py:1]: This was previously setup to be called like an algorithm.  Now that is handled by the `__init__.py`

**Comment by rboston628** [comment]
I ran this with both diffcal and normcal workflows and did not observe any errors

**Review by peterfpeterson — CHANGES_REQUESTED** [review]
I'd prefer that the algorithms themselves kept the `AlgorithmFactory.subscribe()` calls so the algorithm looks like anybody else's. The code in `.../algorithm/__init__.py` is pretty nice and I like the comments in it.

**Pull request merged**